crates/network-scheduler/src/data_chunk.rs

```diff
@@ -60,12 +73,50 @@ impl DataChunk {
 pub fn chunks_to_worker_state(chunks: impl IntoIterator<Item = DataChunk>) -> WorkerState {
     let datasets = chunks
         .into_iter()
-        .map(|chunk| (chunk.dataset_url, chunk.block_range))
+        .map(|chunk| (chunk.dataset_id, chunk.block_range))
         .into_grouping_map()
         .collect();
     WorkerState { datasets }
 }
 
+#[allow(deprecated)]
+pub fn chunks_to_assignment(chunks: impl Iterator<Item = DataChunk>) -> WorkerAssignment {
+    let mut known_filenames: HashMap<String, u32> = HashMap::new();
+    let mut filename_to_index = |filename: String| {
+        let next_index = known_filenames.len() as u32;
+        *known_filenames.entry(filename).or_insert(next_index)
+    };
+    let dataset_chunks = chunks
+        .into_group_map_by(|chunk| chunk.dataset_id.clone())
```

Comment:
I think `into_grouping_map_by` is preferred to `into_group_map_by`, because it's lazy and reduces the amount of map operations. Feel free to update, if you think it matters.

Reply:
In the [documentation](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.into_group_map) it says that it's _essentially a shorthand for `.into_grouping_map().collect::<Vec<_>>()`_

Reply:
Well, but you don't have to `.collect()` before `.map()`, right?

Reply:
You have to, unfortunately. That's how it was implemented before:
https://github.com/subsquid/archive-router/blob/a02b76a0c5bc48fc901227719ece553d0213ae3b/crates/network-scheduler/src/scheduler.rs#L97-L106

Reply:
OK, nevermind.